bin/generate_base_images.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python3
+import getpass
+import shutil
+import subprocess
+import sys
+import tempfile
+from argparse import ArgumentParser, BooleanOptionalAction
+from pathlib import Path
+from typing import List
+
+from jinja2 import Environment, FileSystemLoader
+from truss.contexts.image_builder.util import (
+    truss_base_image_name,
+    truss_base_image_tag,
+)
+
+base_path = Path(__file__).parent.parent
+templates_path = base_path / "truss" / "templates"
+sys.path.append(str(base_path))
+
+
+PYTHON_VERSIONS = {
+    "3.7",
+    "3.8",
+    "3.9",
+}
+
+
+def _bool_arg_str_to_values(bool_arg_str: str) -> List[bool]:
+    if bool_arg_str == "both":
+        return [True, False]
+    if bool_arg_str == "y":
+        return [True]
+    if bool_arg_str == "n":
+        return [False]
+    raise ValueError(
+        "Unexpected str value of bool flag, acceptable values are y/n/both"
+    )
+
+
+def _docker_login():
```

Comment:
For this first pass I've implemented manual base image generation. But it would be good to have a github action for it. I don't think we want to generate these images automatically, but we can have a github action to trigger it manually. Let me take a note to create a github action for it.